python/pyarrow/tests/conftest.py

```diff
@@ -111,11 +117,44 @@ def pytest_configure(config):
 
 
 def pytest_runtest_setup(item):
+    # Apply numpy mark if the test name already contains
```

Comment:
@jorisvandenbossche what do you think about this? I am not sure we really want to do this automatically if the test name contains `numpy`, `pandas` or `ndarray` but it allows us to remove a bunch of marks from `test_array.py`:
See related commit: https://github.com/apache/arrow/pull/41904/commits/e378dd63c0eafc7983d54c8811e4da88a37f4b25

Reply:
Personally, I would say this feels a bit too magical .. It's nice to not have to add all those marks, but I think short term that is more explicit, and longer term I would maybe rather look into splitting test_array.py and putting all tests about conversion to/from numpy in a separate test file, so there are less tests remaining in test_array.py that require a mark.

Reply:
I've decided to remove that and do the CI job only run some specific tests marked as "without_numpy". Those tests will run both with `numpy` and `nonumpy` but we can just select them with `-k without_numpy`.
Those will be just a small selection of the full test_suite.